tests/components/swiss_public_transport/test_service.py

```diff
@@ -0,0 +1,226 @@
+"""Test the swiss_public_transport service."""
+
+import json
+import logging
+from unittest.mock import AsyncMock, patch
+
+from opendata_transport.exceptions import (
+    OpendataTransportConnectionError,
+    OpendataTransportError,
+)
+import pytest
+from voluptuous import error as vol_er
+
+from homeassistant.components.swiss_public_transport.const import (
+    ATTR_CONFIG_ENTRY_ID,
+    ATTR_LIMIT,
+    CONF_DESTINATION,
+    CONF_START,
+    CONNECTIONS_COUNT,
+    CONNECTIONS_MAX,
+    DOMAIN,
+    SERVICE_FETCH_CONNECTIONS,
+)
+from homeassistant.components.swiss_public_transport.helper import unique_id_from_config
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError, ServiceValidationError
+
+from . import setup_integration
+
+from tests.common import MockConfigEntry, load_fixture
+
+_LOGGER = logging.getLogger(__name__)
+
+MOCK_DATA_STEP_BASE = {
+    CONF_START: "test_start",
+    CONF_DESTINATION: "test_destination",
+}
+
+
+@pytest.mark.parametrize(
+    ("limit", "config_data"),
+    [
+        (1, MOCK_DATA_STEP_BASE),
+        (2, MOCK_DATA_STEP_BASE),
+        (3, MOCK_DATA_STEP_BASE),
+        (CONNECTIONS_MAX, MOCK_DATA_STEP_BASE),
+        (None, MOCK_DATA_STEP_BASE),
+    ],
+)
+async def test_service_call_fetch_connections_success(
+    hass: HomeAssistant,
+    limit: int,
+    config_data,
+) -> None:
+    """Test the fetch_connections service."""
+
+    unique_id = unique_id_from_config(config_data)
+
+    config_entry = MockConfigEntry(
+        domain=DOMAIN,
+        data=config_data,
+        title=f"Service test call with limit={limit}",
+        unique_id=unique_id,
+        entry_id=f"entry_{unique_id}",
+    )
+
+    with patch(
+        "homeassistant.components.swiss_public_transport.OpendataTransport",
+        return_value=AsyncMock(),
+    ) as mock:
+        mock().connections = json.loads(load_fixture("connections.json", DOMAIN))[
+            0 : (limit or CONNECTIONS_COUNT) + 2
+        ]
+
+        await setup_integration(hass, config_entry)
+
+        data = {ATTR_CONFIG_ENTRY_ID: config_entry.entry_id}
+        if limit is not None:
```

Comment:
Pass a dict, empty or not empty, as the parameter and update the data instead. Then we don't need this condition.

Reply:
There you go: https://github.com/home-assistant/core/pull/123660